core-kotlin-modules/core-kotlin-lang-oop-2/src/test/kotlin/com/baeldung/sealedclasses/vs/enums/OsEnumUnitTest.kt

```diff
@@ -0,0 +1,32 @@
+package com.baeldung.sealedclasses.vs.enums
+
+import org.junit.jupiter.api.Test
+import kotlin.test.assertEquals
+
+class OsEnumUnitTest {
+    @Test
+    fun detect_os() {
+        OsEnum.values().forEach {
+            detectOsByUsingWhen(it)
+        }
+    }
+
+    private fun detectOsByUsingWhen(osEnum: OsEnum) {
+        assertEquals("Called from parent enum class", osEnum.getTextParent())
+        when (osEnum) {
+            OsEnum.Linux -> assertEquals("Linux by Open-Source - value=1", osEnum.getText(1))
+            OsEnum.Windows -> assertEquals("Windows by Microsoft - value=2", osEnum.getText(2))
+            OsEnum.Mac -> assertEquals("Mac by Apple - released at 2001", osEnum.getText(3))
+            else -> assert(osEnum == OsEnum.Unknown)
```

Comment:
Use assertTrue from kotlin.test, rather than the assert function.
https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/assert-true.html